core/src/avm2/object.rs

```diff
@@ -209,11 +209,14 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
         multiname: &Multiname<'gc>,
         activation: &mut Activation<'_, 'gc>,
     ) -> Result<Value<'gc>, Error<'gc>> {
-        match self.vtable().and_then(|vtable| vtable.get_trait(multiname)) {
-            Some(Property::Slot { slot_id }) | Some(Property::ConstSlot { slot_id }) => {
+        match self
```

Comment:
As written above, this requiring passing `ns` around feels very wrong. Both general design-wise, and perf-wise. Maybe cross-reference avmplus for this? They definitely didn't need to do this kind of thing. The approach in `fn display_method`, while probably also incorrect, managed to avoid it and pushed all work to error time.